analyzers/tests/SonarAnalyzer.UnitTest/Rules/TimestampsShouldNotBeUsedAsPrimaryKeys.cs

```diff
@@ -0,0 +1,58 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.Rules.CSharp;
+
+namespace SonarAnalyzer.UnitTest.Rules;
+
+[TestClass]
+public class TimestampsShouldNotBeUsedAsPrimaryKeysTest
+{
+    private readonly VerifierBuilder builder = new VerifierBuilder<TimestampsShouldNotBeUsedAsPrimaryKeys>();
+
+    [TestMethod]
+    public void TimestampsShouldNotBeUsedAsPrimaryKeys_CSharp() =>
+        builder
+            .AddPaths("TimestampsShouldNotBeUsedAsPrimaryKeys.cs")
+            .AddEntityFrameworkReference()
+            .Verify();
+
+    [TestMethod]
+    public void TimestampsShouldNotBeUsedAsPrimaryKeys_CSharp9() =>
+        builder
+            .AddPaths("TimestampsShouldNotBeUsedAsPrimaryKeys.CSharp9.cs")
+            .AddEntityFrameworkReference()
+            .WithOptions(ParseOptionsHelper.FromCSharp9)
+            .Verify();
+}
+
+internal static class BuilderExtensions
```

Comment:
With my suggestion from above, it will be invoked only once, so it might be a bit of overkill at this point. It could have gone directly on the builder creation, however knowing that VBNet should come at one point, we can leave it as is because it will be reused.
Very nice idea!

Reply:
I moved the `#if` conditions up to the field declaration. This will be duplicated when the VB test cases are introduced, but it looks better for me than having two classes in the same file. Unfortunately extension methods cannot be placed in nested classes.

Reply:
I moved them again to a separate method to better support referencing both EF and EF Core.